src/encode.c

```diff
@@ -29,6 +29,10 @@ int coap_flsll(long long i)
 }
 #endif
 
+/*
+ * The following functions assume little endian systems
```

Comment:
I do not believe that this is true. For all these functions, the number representation in buffer is big endian (= network byte order), and the binary representation is always in host byte order, independent of the actual processor architecture.

Reply:
Yes, you are correct.  Comment removed.